class/Patchwork/Utf8.php

```diff
@@ -566,7 +566,20 @@ static function lcfirst($s)
 
     static function ucwords($s)
     {
-        return mb_convert_case($s, MB_CASE_TITLE, 'UTF-8');
+        if (PHP_VERSION_ID < 50401)
+        {
+            return mb_ereg_replace("\b(.)", "mb_convert_case('\\1', MB_CASE_UPPER, 'UTF-8')", $s, 'e');
```

Comment:
I'll change it to preg_replace_callback. Again the MB_CASE_TITLE vs MB_CASE_UPPER shouldn't matter as it is one character. Also, The original bug here is that ucwords is not supposed to change the case to title case. It is just supposed to change the first character of each word to upper case.